litestar/typing.py

```diff
@@ -165,3 +385,135 @@ def has_inner_subclass_of(self, cl: type[Any] | tuple[type[Any], ...]) -> bool:
+                ``get_type_hints(..., include_extras=True)`` so that forward references are resolved and recursive
+                ``Annotated`` types are flattened.
+            **kwargs: Additional keyword arguments to pass to the ``FieldDefinition`` constructor.
+
+        Returns:
+            FieldDefinition
+        """
+
+        unwrapped, metadata, wrappers = unwrap_annotation(annotation if annotation is not Empty else Any)
+        origin = get_origin(unwrapped)
+
+        args = () if origin is abc.Callable else get_args(unwrapped)
+
+        if not kwargs.get("kwarg_definition"):
+            if isinstance(kwargs.get("default"), (KwargDefinition, DependencyKwarg)):
+                kwargs["kwarg_definition"] = kwargs.pop("default")
+            elif any(isinstance(v, (KwargDefinition, DependencyKwarg)) for v in metadata):
+                kwargs["kwarg_definition"] = [v for v in metadata if isinstance(v, (KwargDefinition, DependencyKwarg))][
+                    0
+                ]
+                metadata = tuple(v for v in metadata if not isinstance(v, (KwargDefinition, DependencyKwarg)))
+            elif (extra := kwargs.get("extra", {})) and "kwarg_definition" in extra:
+                kwargs["kwarg_definition"] = extra.pop("kwarg_definition")
+            else:
+                kwargs["kwarg_definition"], kwargs["extra"] = cls._extract_metada(
+                    annotation=annotation,
+                    name=kwargs.get("name", ""),
+                    default=kwargs.get("default", Empty),
+                    metadata=metadata,
+                    extra=kwargs.get("extra", {}),
+                )
+
+        kwargs.setdefault("annotation", unwrapped)
+        kwargs.setdefault("args", args)
+        kwargs.setdefault("default", Empty)
+        kwargs.setdefault("extra", {})
+        kwargs.setdefault("inner_types", tuple(FieldDefinition.from_annotation(arg) for arg in args))
+        kwargs.setdefault("instantiable_origin", get_instantiable_origin(origin, unwrapped))
+        kwargs.setdefault("kwarg_definition", None)
+        kwargs.setdefault("metadata", metadata)
+        kwargs.setdefault("name", "")
+        kwargs.setdefault("origin", origin)
+        kwargs.setdefault("raw", annotation)
+        kwargs.setdefault("safe_generic_origin", get_safe_generic_origin(origin, unwrapped))
+        kwargs.setdefault("type_wrappers", wrappers)
+
+        instance = FieldDefinition(**kwargs)
+        if not instance.has_default and instance.kwarg_definition:
+            return replace(instance, default=instance.kwarg_definition.default)
+
+        return instance
+
+    @classmethod
+    def from_kwarg(
+        cls,
+        annotation: Any,
+        name: str,
+        default: Any = Empty,
+        inner_types: tuple[FieldDefinition, ...] | None = None,
+        kwarg_definition: KwargDefinition | DependencyKwarg | None = None,
+        extra: dict[str, Any] | None = None,
+    ) -> FieldDefinition:
```

Comment:
This looks like a restricted wrapper of `from_annotation`. Is it really needed? How does one choose which to use?